### ISSUE TITLE
very bad lag spikes due to pathfinding

### ISSUE THREAD

**Issue by shmooe** [OpenRA/OpenRA#4584]
After about 10 or 15 mins the game starts to freeze and play fine every few secs. I turned on the graph and the (white) pathingfind and (darkblue) Tick_time start spiking and eventually start going off my screen. none of my components are maxing. Was is the cause so i can turn it off or fix it. I have a intel i5 radeon 280x 16gb ram win 7 ult

**Comment by lebedkin** [comment]
Hi, sorry for my English(
the problem is still exist. 
![screen shot 2015-04-25 at 7 04 19 pm](https://cloud.githubusercontent.com/assets/1270800/7353456/4eb22c4e-ecd9-11e4-9245-38de978a2383.png)
in skirmish mod of Dune 2000 after ~10 minutes of play time the game starts lagging every 1-2 seconds. According the performance graphic it's tick_time.
I don't have a video but I have a replay of the game https://www.dropbox.com/s/jsgn92wuwe0h5yk/OpenRA-2015-04-25T230024Z.orarep?dl=0

**Comment by Mailaender** [comment]
Did you block the path for your harvesters or the bot attack routes?

**Comment by obrakmann** [comment]
Those spikes are tick_time, not pathfinding, though.  The 1 sec interval *might* point to the rescan interval in EnemyWatcher.cs, but mainly because that's the only place I currently know with a 1 sec interval ;)

**Comment by Mailaender** [comment]
The perf.log will reveal the problem.

**Comment by RoosterDragon** [comment]
The logs seem to suggest it's a pathfinding issue, lots of slow ticks for move related traits.

I imagine something is trying to path somewhere degenerate, thus causing a fairly long search, and when it fails it's retrying every second without much luck.

**Comment by pchote** [comment]
I saw this on too, but on a much smaller timescale:
```
   260 ms [7357] Activity: Move
   260 ms [7357] Activity: AttackMoveActivity
   229 ms [7357] Activity: Move
   229 ms [7357] Activity: AttackMoveActivity
   382 ms [7357] Activity: Move
   382 ms [7357] Activity: AttackMoveActivity
   312 ms [7359] Activity: Move
   312 ms [7359] Activity: AttackMoveActivity
   235 ms [7359] Activity: Move
   235 ms [7359] Activity: AttackMoveActivity
   210 ms [7359] Activity: Move
   210 ms [7359] Activity: AttackMoveActivity
   332 ms [7363] Activity: Move
   332 ms [7363] Activity: AttackMoveActivity
   288 ms [7366] Activity: Move
   288 ms [7366] Activity: AttackMoveActivity
```
and so on (this is only a very small snippet).

**Comment by pchote** [comment]
Replay file (recorded on 1a71e6d8f8de51da7bdb1f821acc77fcd768a40a in #7904): https://www.dropbox.com/s/83i2uwkzoe3tndk/OpenRA-2015-04-28T145728Z.orarep?dl=0

**Comment by LipkeGu** [comment]
i had played two AIs ... and everytime when they choose the target then the game will lag shortly... sith these spikes =(

**Comment by MYRWYR** [comment]
Hi folks, I'm encountering this tick spike in 202103-Release, on a small map with one AI opponent. Have enabled the stats overlay to confirm the ticks peak far higher than the top of the graph.

Edit: I am in fact using the ORA Shattered Paradise mod, it might be an issue there instead. Is there any way I can confirm whether or not SP is the problem?

**Comment by Mailaender** [comment]
You can confirm that this is a not a problem anymore in https://github.com/OpenRA/OpenRA/releases/tag/playtest-20221224

**Comment by MYRWYR** [comment]
> You can confirm that this is a not a problem anymore in https://github.com/OpenRA/OpenRA/releases/tag/playtest-20221224

I've messed around and "resolved" it - destroying a specific building made the ticks drop down to normal immediately.

Edit: I have confirmed via the replay that several small infantry units were in fact stuck behind the building, in a little crevice. It seems they moved there when the AI placed the building on top of them. Whenever the AI tries to attack, the units get stuck in a pathfinding loop.
